### PR TITLE
Bug 1179199 - [Stingray][Notification] Establish IAC with smart-system

### PR DESCRIPTION
**Pull request by rexboy7**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f9c408898a8f9b6361aa08cd51215a2557789721

File: tv_apps/smart-system/js/notification_connector.js [hunk 1]
```diff
@@ -1,21 +1,80 @@
...
+  onIACMessage: function(message) {
+    var detail = JSON.parse(message.data);
+    switch(detail.type) {
+    case 'close-notification':
+      console.error('Received close message event:' + detail.id);
+      break;
+    case 'click-notification':
+      console.error('Received click message event' + detail.id);
+      break;
+    case 'added-notification':
```
Comment (begeeben) [inline tv_apps/smart-system/js/notification_connector.js]: Just a tiny nit. The tenses of these verbs don't seem to be consistent.

File: tv_apps/smart-system/js/notification_connector.js [hunk 2]
```diff
@@ -1,21 +1,80 @@
...
+      return Promise.resolve(this._port);
+    }
+
+    return new Promise(function(resolve, reject) {
+      navigator.mozApps.getSelf().onsuccess = function(evt) {
+        var app = evt.target.result;
+        app.connect('notification-message').then(function onAccepted(ports) {
+          if (ports.length !== 1) {
+            console.error('notification service: more than one app found?');
+            reject();
```
Comment (john-hu) [inline tv_apps/smart-system/js/notification_connector.js]: forget to add `return` here.

File: tv_apps/smart-system/js/notification_connector.js (hunk 1 above)
Comment (john-hu) [inline tv_apps/smart-system/js/notification_connector.js]: agree

File: tv_apps/smart-system/js/notification_connector.js [hunk 3]
```diff
@@ -1,21 +1,80 @@
...
-      ports.forEach(function(port) {
-        // XXX: Testing purpose. Please remove it once real notification sending
-        // service is done.
-        port.postMessage('hello');
-      });
-    }, function onConnRejected(reason) {
-      console.log('rejected');
+  window.addEventListener('notification-message', this);
+  window.addEventListener('desktop-notification-resend', this);
+  connect();
```
Comment (john-hu) [inline tv_apps/smart-system/js/notification_connector.js]: `connect` is not defined. You should use `this.connect`.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=59e43d477bcd4bc9561b5e4f5b337625ccd98136

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6353137f312c720d19ab5cc243aeef41b71cf058

**Pull request merged**